### PR TITLE
fix(a380x/air cond): wasm crash during rapid decompression

### LINKED ISSUE #9215: Doors Open and Airplane+System Freeze
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-10-31T14:33:02+00:00",
    "ref": "refs/tags/v0.12.0",
    "sha": "f26029363ad3edeeafbed20aa79c13b32c8090f8",
    "actor": "FoxtrotSierra6829",
    "event_name": "manual",
    "pretty_release_name": "stable/v0.12.0",
    "version": "a380x-v0.12.0-HEAD.f260293"
}
```

### Describe the bug

Hey, I was flying to Dubai when suddenly, the doors opened mid-air over the Red Sea. I managed to close them by changing the page on the EFB. On Volanta, I saw that I was off course, even though I hadn’t changed anything. The PFD was frozen too. I couldn’t take over control by hand, and the plane didn’t respond to anything. My friend had exactly the same bug.

### Expected behavior

That the airplane continues to fly normally ;)

### Steps to reproduce

Video Link: https://youtu.be/DSDSrQeMCQ4

Video from my friend: https://youtu.be/X_B7gLpjw2c

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

TxmoTK

### PR DESCRIPTION
**Pull request by mjuhe**
Fixes #9215

## Summary of Changes

Fixes an issue with the cabin air flow demand calculation that was causing a wasm crash during rapid decompression.

## Additional context

The main issue was with the way the pack demand flow and the cabin fan extraction flow was calculated (some values went negative, which drove other values to infinity). I've also added a couple of small fixes that should help in case of adirs failures in flight.

Discord username (if different from GitHub): Commander Ga

## Testing instructions

1. Climb to FL400
2. Trigger the "Rapid Decompression" faliure on the EFB
3. Validate that the cabin altitude gets to ~40000ft and that the cabin temperatures stay in a reasonable range (20-30 deg)
4. Fly for a few minutes, validate the wasm doesn't crash (the PFD continues to operate) and the cabin vertical speed stays within a reasonable range (+/- 100fpm)

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — APPROVED** [review]

File: fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs [hunk 1]
```diff
@@ -979,19 +979,21 @@ impl<C: PressurizationConstants> CabinPressureControlSystemApplication<C> {
     ) {
         let (adirs_airspeed, adirs_ambient_pressure) = self.adirs_values_calculation(adirs);
         self.exterior_airspeed = adirs_airspeed.unwrap_or_default();
+        let new_exterior_altitude: Length;
 
-        self.exterior_flight_altitude = if !self.is_initialised {
+        if !self.is_initialised && adirs_ambient_pressure.is_some() {
```
Comment (Gurgel100) [inline fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs:984]: Please assign `new_exterior_altitude` here.
Also just to make sure: this branch can only be taken when it's not initialised and the adirs are available.

**Comment by alexr4339** [comment]
Quality Assurance Tester

Discord Username  : alexdnzdmb
Object of testing : #9543 
Aircraft          :A380X
Tier of Testing   : 1
Date              : 17/11/2024

Testing Process: 
Climb to FL400 - check
Trigger the "Rapid Decompression" faliure on the EFB - check
Validate that the cabin altitude gets to ~40000ft and that the cabin temperatures stay in a reasonable range (20-30 deg) - check, increased with around 6400 (seems as if 6400fpm is max indicated value) cabin fpm until probably 40k. However value only showed cab alt of 30k feet but cab alt still increased from 30k onwards until it probably reached around 40k, delta PSI indicates 0 as well
image
image

Fly for a few minutes, validate the wasm doesn't crash (the PFD continues to operate) and the cabin vertical speed stays within a reasonable range (+/- 100fpm) - check

Testing Results: 
Not passed (as indicated values are incorrect), passed if that is not scope of PR

Negatives:       
CAB alt only shows values up to 30k feet amd doesnt match the cab alt clock instrument indication

Conclusions: 


Media:

File: fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs (hunk 1 above)
Comment (mjuhe) [reply fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs]: Yes you are right, this condition is not very useful, it was probably me trying to be overly safe when making sure there was no condition where the reference pressure was 0. I reverted these changes now.

**Comment by mjuhe** [comment]
Thanks for the test @alexr4339 !! Sorry I forgot to mention that the indications only go to +/- 6400fpm (on the vertical speed) and 30,000ft (on the cabin altitude). This is a feature, not a bug 😄as that's how the ARINC data gets transmitted. You can see the real value of the cabin altitude checking the `A32NX_PRESS_MAN_CABIN_ALTITUDE` LVar in Behaviours.

Although if you experienced no unexpected jumps in VS or in cabin temperature, and most importantly no wasm crashes, I don't think this needs to be tested again.

**Comment by flogross89** [comment]
can we merge this?

**Comment by mjuhe** [comment]
> can we merge this?

All good from my side, will merge.

**Review by Gurgel100 — APPROVED** [review]

**Pull request merged**